Description: Patch to handle utf8 with RRDTool version >= 1.3.  Not fixed in
 upstream tags/1.3.4, and trunk is rewritten.
Origin: other, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=494547
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=494547
Forwarded: no
Index: munin-1.2.6/server/munin-graph.in
===================================================================
--- munin-1.2.6.orig/server/munin-graph.in	2008-09-08 21:42:45.000000000 +0200
+++ munin-1.2.6/server/munin-graph.in	2008-09-08 21:52:01.000000000 +0200
@@ -30,6 +30,7 @@
 use Digest::MD5;
 use Getopt::Long;
 use Time::HiRes;
+if ($RRDs::VERSION >= 1.3) { use Encode; }
 
 my $graph_time= Time::HiRes::time;
 my $DEBUG = 0;
@@ -858,6 +859,16 @@
 	    }
 	    print "\n\nrrdtool \"graph\" \"",
 	      join ("\"\n\t\"",@complete), "\"\n" if $DEBUG;
+
+	    # Since version 1.3 rrdtool uses libpango which needs its input  
+	    # as utf8 string. So we assume that every input is in latin1 
+	    # and decode it to perl's internal representation and then to utf8. 
+     	    if ( $RRDs::VERSION >= 1.3 ) { 
+ 	        @complete = map { 
+		    $_ = encode("utf8", (decode("latin1", $_)));
+		} @complete;
+	    }
+
 	    RRDs::graph (@complete);
 	    if (my $ERROR = RRDs::error) {
 		logger ("Unable to graph $filename: $ERROR");
